src/cpu/core_dyn_x86/decoder.h

```diff
@@ -2896,7 +2896,9 @@ static CacheBlock * CreateCacheBlock(CodePageHandler * codepage,PhysPt start,Bit
 	/* Setup the correct end-address */
 	decode.active_block->page.end=--decode.page.index;
 	dyn_mem_execute(cache_addr, cache_bytes);
-	dyn_cache_invalidate(cache_addr, cache_bytes);
+	const size_t cache_flush_bytes = static_cast<size_t>(decode.block->cache.size);
+	dyn_cache_invalidate(cache_addr, cache_flush_bytes);
+	assert(decode.block->cache.size <= cache_bytes);
```

Comment:
(question for my own understanding)
Should the assert come before the invalidation?  Or does invalidating the decoded block reduce the cache by the block's size? (in which case, asserting after makes sense?)

Reply:
In this case, I wanted the last thing before returning to be a sanity check that we didn't create a cache block whose size exceed `cache_bytes` i.e. CACHE_MAXSIZE.